src/test/java/com/google/sps/CareerQuizTest.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps;
+
+import com.google.sps.servlets.CareerQuizServlet;
+import com.google.sps.data.CareerQuestionDatabase;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.sps.data.CareerQuestionChoice;
+import com.google.sps.data.CareerQuestionAndChoices;
+import static org.mockito.Mockito.*;
+
+
+/** Tests the CareerQuiz servlet and its interactions with CareerQuestionDatabase*/
+@RunWith(JUnit4.class)
+public final class CareerQuizTest {
```

Comment:
So use can use [GuiceBerry](https://github.com/zorzella/guiceberry) to do dep injection in your tests.

However, I actually like the idea of moving the following into a method, and mocking that. Then you don't need guice at all :)

```
ArrayList<CareerQuestionAndChoices> questionsAndChoices = 
        this.careerQuestionDatabase.getQuestionsAndChoices()
```

Reply:
Actually, you might not even need GuiceBerry:

Here is some code I wrote yesterday:
```
  @Inject private FixedWidthTestFile testFile;
  @Inject private WellsFargoNachaRequestFileParserImpl parser;

  @Before
  public void injectMembers() {
    Guice.createInjector(new WellsFargoRequestFileParserModule()).injectMembers(this);
  }
```

Reply:
So you might be able to leverage this. Here is a great example:
```
// Class under test.
public class MyClass {

  // Guice-injected field
  @Inject
  private HelperService helper;

  int methodToTest() {
    return helper.getIntValue();
  }
}

@RunWith(JUnit4.class)
public class MyClassTest {

  // Create a mock helper service that will be injected into MyClass
  @Bind @Mock private HelperService mockHelperService;

  // Class to test, injected via Guice, that has mockHelperService as
  // its dependency
  @Inject private MyClass myClass;

  @Before
  public void createInjector() {
    // BoundFieldModule finds all fields that are annotated with @Bind and makes
    // them available for use as dependencies, then instantiates and injects anything
    // annotated with @Inject.
    Guice.createInjector(BoundFieldModule.of(this)).injectMembers(this);
  }

  @Test
  public void methodToTestReturnsValue() {
    // Use our mock to specify a return value
    when(mockHelperService.getIntValue()).thenReturn(1);

    // Get the specified return value.
    assertThat(myClass.methodToTest()).isEqualTo(1);
  }
}
```

Reply:
https://google.github.io/guice/api-docs/4.1/javadoc/com/google/inject/testing/fieldbinder/BoundFieldModule.html

And the mock is from `import org.mockito.Mock;`

Reply:
other imports:
```
import com.google.inject.testing.fieldbinder.Bind;
import com.google.inject.testing.fieldbinder.BoundFieldModule;
```